lib/node_modules/@stdlib/math/base/special/wrapf/src/main.c

```diff
@@ -0,0 +1,70 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/wrapf.h"
+#include "stdlib/math/base/assert/is_nanf.h"
+#include "stdlib/math/base/special/truncf.h"
+#include "stdlib/math/base/assert/is_negative_zerof.h"
+#include <math.h>
+
+/**
+* Wraps a value on the half-open interval [min,max) for single-precision floating-point numbers.
+*
+* @param v    input value
+* @param min  minimum value
+* @param max  maximum value
+* @return     wrapped value
+*
+* @example
+* float v = stdlib_base_wrapf( 3.14, 0.0, 5.0 );
+* // returns 3.14
+*/
+float stdlib_base_wrapf( const float v, const float min, const float max ) {
+	float delta;
+	float maxc;
+	float minc;
+	float vc;
+
+	if ( stdlib_base_is_nanf( v ) || stdlib_base_is_nanf( min ) || stdlib_base_is_nanf( max ) || max <= min ) {
+		return 0.0 / 0.0; // NaN
+	}
+	maxc = max;
+	minc = min;
+	vc = v;
+
+	// Normalize +-0 to +0...
+	if ( stdlib_base_is_negative_zerof( vc ) ) {
+		vc = 0.0;
+	}
+	if ( stdlib_base_is_negative_zerof( minc ) ) {
+		minc = 0.0;
+	}
+	if ( stdlib_base_is_negative_zerof( maxc ) ) {
+		maxc = 0.0;
+	}
+	// Simple case where value is already within range...
+	if ( minc <= vc && vc < maxc ) {
+		return vc;
+	}
+	// Perform range reduction...
+	delta = maxc - minc;
+	if ( vc < minc ) {
+		vc += delta * ( stdlib_base_truncf( ( minc - vc ) / delta ) + 1.0 );
+	}
+	return minc + ( fmodf( vc - minc, delta ) );
```

Comment:
@gururaj1512 The `fmodf` which is being used here, is not being included at the top. We are currently missing the implementation of `fmodf`. You'll need to include it once we have added `fmodf`.

Reply:
@gunjjoshi, Thanks for clarifying

Reply:
@gururaj1512 `fmodf` (added by @gunjjoshi in https://github.com/stdlib-js/stdlib/pull/3059) has now landed, so this branch can be rebased on the latest code on `develop` and updated to make use of it.